spec/unit/lib/temporal/workflow/context_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'temporal/workflow'
+require 'temporal/workflow/context'
+
+class MyTestWorkflow < Temporal::Workflow; end
+
+describe Temporal::Workflow::Context do
+  let(:state_manager) { instance_double('Temporal::Workflow::StateManager') }
+  let(:dispatcher) { instance_double('Temporal::Workflow::Dispatcher') }
+  let(:metadata) { instance_double('Temporal::Metadata::Workflow') }
+  let(:workflow_context) {
+    Temporal::Workflow::Context.new(state_manager, dispatcher, MyTestWorkflow, metadata, Temporal.configuration)
+  }
+
+  describe '#upsert_search_attributes' do
+    it 'does not accept nil' do
+      expect do
+        workflow_context.upsert_search_attributes(nil)
+      end.to raise_error(ArgumentError, 'search_attributes cannot be nil')
+    end
+
+    it 'requires a hash' do
+      expect do
+        workflow_context.upsert_search_attributes(['array_not_supported'])
+      end.to raise_error(ArgumentError, 'for search_attributes, expecting a Hash, not Array')
+    end
+
+    it 'requires a non-empty hash' do
+      expect do
+        workflow_context.upsert_search_attributes({})
+      end.to raise_error(ArgumentError, 'Cannot upsert an empty hash for search_attributes, as this would do nothing.')
+    end
+
+    it 'creates a command to execute the request' do
+      allow(state_manager).to receive(:schedule)
+        .with an_instance_of(Temporal::Workflow::Command::UpsertSearchAttributes)
```

Comment:
`allow` doesn't set the expectation, just the stub. You need to add `expect().to have_received().with()` to actually ensure that it was correctly sent

Reply:
Whoops!